views/my_packages.package.dt

```diff
@@ -19,6 +21,36 @@ block body
 				|= e.get!string
 				br
 
+	h2 General
+
+	.inputForm
+		form(method="POST", enctype="multipart/form-data", action="#{req.rootDir}my_packages/#{packageName}/set_logo")
+			h3 Logo
+			- string existing = pack["logo"].get!string;
+			- string logoPath = existing.length ? (existing ~ "?" ~ uniform(ulong.min, ulong.max).to!string) : (req.rootDir ~ "images/default-logo.png");
```

Comment:
This looks like a hack around sending proper caching headers or using checksums as part of the file path.

Reply:
Otherwise directly after uploading you would still see the old path. But yeah this should rather contain some hash or date. Proper caching isn't really possible because we want to have fast caching without checking the server if it is modified here because logos rarely change. That's why the cache reload delay is set to 7 days too